src/DataGrid.tsx

```diff
@@ -771,7 +771,7 @@ function DataGrid<R, SR, K extends Key>(
     if (onSelectedCellChange && !samePosition) {
       onSelectedCellChange({
         rowIdx: position.rowIdx,
-        row,
+        row: isRowIdxWithinViewportBounds(position.rowIdx) ? rows[position.rowIdx] : undefined,
         column: columns[position.idx]
```

Comment:
Maybe we can change the type to include row type similar to 
https://github.com/adazzle/react-data-grid/blob/main/src/types.ts#L284